sympy/core/numbers.py

```diff
@@ -768,7 +771,20 @@ class Float(Number):
 
     is_Float = True
 
-    def __new__(cls, num, prec=None):
+    def __new__(cls, num, dps=None, prec=None, precision=None):
+        if dps and precision or dps and prec or prec and precision is not None:
```

Comment:
maybe you should move the `prec` deprecation stuff before this test so there is not `prec` in this if?